### ISSUE TITLE
TC002 false positive when using import in except

### ISSUE THREAD

**Issue by kurtschelfthout** [snok/flake8-type-checking#101]
The following triggers TC002

```python
import botocore.exceptions

if __name__ == "__main__":
    try:
        raise Exception("test")
    except botocore.exceptions.ClientError:
        print("got botocore error")
    except Exception:
        print("got other error")
```

Unfortunately if you action that advice, the code will fail at runtime in the except block.

**Comment by sondrelg** [comment]
Thanks for opening the issue. I'll create a fix for this this week most likely. Seems like an easy fix 😊

**Comment by sondrelg** [comment]
Looks like 

```python
import urllib.parse 

urllib.parse.urlencode('')
```

Also triggers the same issue. We weren't handling double imports like this properly I think. 

Would you mind testing if the code in https://github.com/snok/flake8-type-checking/pull/103 solves your use-case @kurtschelfthout? 🙂 You should be able to install the branch by using `pip install git+https://github.com/snok/flake8-type-checking.git#except-block` (or `poetry add --dev git+https://github.com/snok/flake8-type-checking.git#except-block`)

**Comment by sondrelg** [comment]
If I'm not mistaken, this should also fix #102 🤞

**Comment by kurtschelfthout** [comment]
Can confirm both are fixed.

Beautiful!
